web/app/js/components/NetworkGraph.jsx

```diff
@@ -0,0 +1,216 @@
+          _.map(rows, row => {
+            links.push({
+              source: row.resource.name,
+              target: dst
+            });
+            nodeList.push(row.resource.name);
+            nodeList.push(dst);
+          });
+        });
+
+        let nodes = _.map(_.uniq(nodeList), n => {
+          return { id: n, r: 15 };
+        });
+
+        this.setState({
+          nodes: nodes,
+          links: links,
+          pendingRequests: false,
+          error: ""
+        });
+      })
+      .then( () => this.updateGraph())
+      .catch(this.handleApiError);
+  }
+
+  updateGraph() {
+    this.svg.append("svg:defs").selectAll("marker")
+      .data(["end"])      // Different link/path types can be defined here
+      .enter().append("svg:marker")    // This section adds in the arrows
+      .attr("id", String)
+      .attr("viewBox", "0 -5 10 10")
+      .attr("refX", 24)
+      .attr("refY", -0.25)
+      .attr("markerWidth", 3)
+      .attr("markerHeight", 3)
+      .attr("orient", "auto")
+      .append("svg:path")
+      .attr("d", "M0,-5L10,0L0,5");
+
+    // add the links and the arrows
+    const path = this.svg.append("svg:g").selectAll("path")
+      .data(this.state.links)
+      .enter().append("svg:path")
+      .attr("stroke-width", 3)
+      .attr("stroke", "#000000")
+      .attr("marker-end", "url(#end)");
+
+    const nodeElements = this.svg.append('g')
+      .selectAll('circle')
+      .data(this.state.nodes)
+      .enter().append('circle')
+      .attr("r", 15)
+      .attr('fill', 'steelblue')
+      .call(d3.drag()
+        .on("start", this.dragstarted)
+        .on("drag", this.dragged)
+        .on("end", this.dragended));
+
+    const textElements = this.svg.append('g')
+      .selectAll('text')
+      .data(this.state.nodes)
+      .enter().append('text')
+      .text(node => node.id)
+      .attr('font-size', 15)
+      .attr('dx', 15)
+      .attr('dy', 4);
+
+    simulation.nodes(this.state.nodes).on("tick", () => {
+      path
+        .attr("d", node =>  "M" +
+              node.source.x + " " +
+              node.source.y + " L " +
+              node.target.x + " " +
+              node.target.y);
+      nodeElements
+        .attr("cx", node => node.x)
+        .attr("cy", node => node.y);
+
+      textElements
+        .attr("x", node => node.x)
+        .attr("y", node => node.y);
+    });
+
+    simulation.force("link")
+      .links(this.state.links);
+  }
+
+  dragstarted(d) {
+    if (!d3.event.active) {simulation.alphaTarget(0.3).restart();}
+    d.fx = d.x;
+    d.fy = d.y;
+  }
+
+  dragged(d) {
+    d.fx = d3.event.x;
+    d.fy = d3.event.y;
+  }
+
+  dragended(d) {
+    if (!d3.event.active) {simulation.alphaTarget(0);}
+    d.fx = null;
+    d.fy = null;
+  }
+
+  handleApiError(e) {
+    if (e.isCanceled) {
+      return;
+    }
+
+    this.setState({
+      pendingRequests: false,
+      error: `Error getting data from server: ${e.message}`
```

Comment:
Could this be updated with *what* failed? Maybe `Error getting data required for network graph, please try again later`.